gobblin-compaction/src/main/java/gobblin/compaction/mapreduce/MRCompactorJobPropCreator.java

```diff
@@ -197,10 +202,19 @@ protected MRCompactorJobPropCreator(Builder<?> builder) {
     return Optional.of(dataset);
   }
 
+  private Object shouldDeduplicate() {
```

Comment:
You may not need this method, since it is only one line and used only once. Also, it didn't cover the case in `addInputLateFilesForFirstTimeCompaction`.